### PR TITLE
Fix import order for `if TYPE_CHECKING:` block

### PR DESCRIPTION
**Pull request by DanielYang59**
## Summary

Major changes:

- Fix import order for `if TYPE_CHECKING:` block: put this type checking import block at the end

## Summary by CodeRabbit

- **Refactor**
	- Adjusted import statements and order related to `TYPE_CHECKING` across multiple modules for better code organization.
	- Removed unnecessary lines and added new lines for clarity in file formatting.

- **Chores**
	- Updated import statements for type hints and external libraries in several modules.

- **Style**
	- Improved code style by reorganizing import statements and modifying whitespace.

- **Tests**
	- Enhanced testing capabilities by adding an import statement for settings loading in the test suite.

### CODE REVIEW

**Comment by DanielYang59** [comment]
Meanwhile I noticed our linter `ruff` seems unable to detect such format issue?

There is another issue: the import block seems to have inconsistent format across files, some have two blank lines between the code body, while some have only one?

**Comment by DanielYang59** [comment]
Can you please review this as well? @janosh Thanks a lot!

File: pymatgen/io/ase.py [hunk 1]
```diff
@@ -38,6 +31,13 @@ def __init__(self, *args, **kwargs):
             raise no_ase_err
```
Comment (janosh) [inline pymatgen/io/ase.py:33]: i would stick to single blank line above `if TYPE_CHECKING:`. would be nice if `ruff` enforced this

Comment (DanielYang59) [reply pymatgen/io/ase.py:33]: I tried to remove the extra blank line but `ruff` added it back. `ruff` seem to have a weird interaction with `if TYPE_CHECKING:` for some reason?

**Review by janosh — APPROVED** [review]

File: pymatgen/io/ase.py (hunk 1 above)
Comment (DanielYang59) [reply pymatgen/io/ase.py:33]: Thanks for reviewing. I guess for such issues we'd better wait until `ruff` has some solution (or perhaps already has but we don't know about), and run `ruff` to fix all issues.

**Pull request merged**